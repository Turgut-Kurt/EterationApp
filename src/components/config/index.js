import {fonts, sizes, weights} from './fonts';

import PropTypes from 'prop-types';
import {colors} from './colors';
import {gs} from './gs';

export {PropTypes,fonts, sizes, weights, colors, gs};
