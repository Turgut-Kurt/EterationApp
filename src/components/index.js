// exports All config
export * from './config';

// exports All components
export * from './CustomText';
export * from './CustomButton';
export * from './CustomInputLabel';
export * from './CustomHeader';
export * from './CustomHeader2';
