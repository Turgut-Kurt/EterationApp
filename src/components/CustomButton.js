import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {calcWidth, fontSize} from '~/utils';
import {colors, gs, sizes} from './config';

import {CustomText} from './CustomText';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import React from 'react';

const CustomButton = props => {
  const {
    style,
    children,
    withIcon,
    bordered,
    soloIcon,
    iconColor,
    bgColor,
    textColor,
    name,
    ...rest
  } = props;
  const {
    DefaultButtonStyle,
    BorderedButtonStyle,
    WithIconStyle,
    SoloIconStyle,
  } = styles;
  return (
    <TouchableOpacity
      style={StyleSheet.flatten([
        DefaultButtonStyle,
        StyleSheet.flatten([style]),
        bordered && StyleSheet.flatten(BorderedButtonStyle),
        withIcon && StyleSheet.flatten(WithIconStyle),
        !!soloIcon && StyleSheet.flatten(SoloIconStyle),
        bgColor && StyleSheet.flatten({backgroundColor: bgColor}),
      ])}
      {...rest}>
      {!soloIcon && (
        <CustomText
          style={[
            {marginRight: 5},
            !bordered && {color: colors.color7},
            textColor && {color: textColor},
          ]}
          children={children}
        />
      )}

      {withIcon && (
        <MaterialCommunityIcons
          name={name}
          color={!!iconColor ? iconColor : colors.color7}
          size={sizes.f26}
        />
      )}
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  DefaultButtonStyle: {
    width: calcWidth(100) - 40,
    ...gs.asc,
    height: fontSize(50),
    backgroundColor: colors.color9,
    borderRadius: 10,
    ...gs.jccaic,
    ...gs.shadow2,
  },
  BorderedButtonStyle: {
    width: calcWidth(100) - 40,
    ...gs.asc,
    height: fontSize(46),
    backgroundColor: colors.color7,
    borderWidth: 2,
    borderColor: colors.mainColor,
    borderRadius: 10,
    ...gs.jccaic,
  },
  WithIconStyle: {
    backgroundColor: colors.color9,
    ...gs.fdr,
    ...gs.asc,
    ...gs.jccaic,
    width: 'auto',
    paddingHorizontal: 10,
  },
  SoloIconStyle: {
    backgroundColor: colors.color7,
    borderWidth: 2,
    borderColor: colors.color9,
  },
});

export {CustomButton};
