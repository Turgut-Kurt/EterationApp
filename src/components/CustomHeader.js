import {StyleSheet, TouchableOpacity, View} from 'react-native';
import {calcWidth, fontSize, goBack} from '~utils';
import {colors, gs} from '~/components/config';

import {CustomText} from './CustomText';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import React from 'react';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

const CustomHeader = props => {
  const insets = useSafeAreaInsets();
  const {title} = props;
  return (
    <View
      style={[
        styles.Container,
        {height: fontSize(50) + insets.top, paddingTop: insets.top},
      ]}>
      <TouchableOpacity onPress={goBack} style={styles.LeftContainer}>
        <MaterialIcons
          style={styles.IconStyle}
          name="chevron-left"
          size={30}
          color={colors.color7}
        />
      </TouchableOpacity>
      <View style={styles.CenterContainer}>
        <CustomText
          fSemibold
          style={{color: colors.color7}}
          f20
          children={title}
        />
      </View>
      {props.rightContainer ? (
        props.rightContainer
      ) : (
        <View style={styles.RightContainer} />
      )}
    </View>
  );
};
const styles = StyleSheet.create({
  Container: {
    ...gs.jccaic,
    ...gs.fdr,
    ...gs.posRel,
    backgroundColor: colors.mainColor,
    zIndex: -1,
  },
  LeftContainer: {
    flex: 0.15,
    ...gs.jcc,
    height: '100%',
  },
  IconStyle: {...gs.posAbs, left: calcWidth(6)},
  CenterContainer: {
    flex: 0.7,
    height: '100%',
    ...gs.jccaic,
  },
  RightContainer: {
    flex: 0.15,
    height: '100%',
    ...gs.jccaic,
  },
});
export {CustomHeader};
