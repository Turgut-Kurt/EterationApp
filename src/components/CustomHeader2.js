import {StyleSheet, View} from 'react-native';
import {colors, gs} from '~/components/config';

import {CustomText} from './CustomText';
import React from 'react';
import {fontSize} from '~utils';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

const CustomHeader2 = props => {
  const insets = useSafeAreaInsets();
  const {title} = props;
  return (
    <View
      style={[
        styles.Container,
        {height: fontSize(50) + insets.top, paddingTop: insets.top},
      ]}>
      <View style={styles.CenterContainer}>
        <CustomText
          fSemibold
          style={{color: colors.color7}}
          f20
          children={title}
        />
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  Container: {
    ...gs.jccaic,
    ...gs.fdr,
    ...gs.posRel,
    backgroundColor: colors.mainColor,
  },
  CenterContainer: {
    flex: 1,
    height: '100%',
    ...gs.jccaic,
  },
});
export {CustomHeader2};
