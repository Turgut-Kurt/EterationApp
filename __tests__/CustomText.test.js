import {fireEvent, render, screen} from '@testing-library/react-native';

import {CustomText} from '../src/components/CustomText';
import React from 'react';

test('should match with snapshot', () => {
  const comp = render(<CustomText children={'Test'} />);
  expect(comp).toMatchSnapshot();
});

test('should render text correctly', () => {
  const testText = 'test';
  const comp = render(<CustomText children={testText} />);

  const customText = comp.getByTestId('custom-text');
  expect(customText).toBe(testText);
});
