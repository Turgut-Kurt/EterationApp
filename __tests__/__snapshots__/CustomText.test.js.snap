// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#1F2232",
      "fontFamily": "EncodeSans-Regular",
      "fontSize": 28,
      "paddingTop": 1,
    }
  }
  testID="custom-text-children"
>
  Test
</Text>
`;
